it/src/main/java/com/google/cloud/teleport/it/mongodb/MongoDBResourceManagerUtils.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (C) 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.it.mongodb;
+
+import static com.google.cloud.teleport.it.common.ResourceManagerUtils.generateResourceId;
+
+import com.google.re2j.Pattern;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Utilities for {@link com.google.cloud.teleport.it.mongodb.MongoDBResourceManager}
+ * implementations.
+ */
+final class MongoDBResourceManagerUtils {
+
+  private static final int MAX_DATABASE_NAME_LENGTH = 63;
+  private static final Pattern ILLEGAL_DATABASE_NAME_CHARS =
+      Pattern.compile("[\\/\\\\. \"\0$]"); // i.e. [/\. "$]
```

Comment:
nit Do we have a source/link to cite the illegal chars? Good for posterity

Reply:
ack.